app/assets/javascripts/qonsole_rails/qonsole.js

```diff
@@ -90,6 +95,11 @@ modulejs.define( "qonsole", [
   var showGivenQuery = function( query ) {
     if (query === "_localstore") {
       query = Util.Browser.getSessionStore( "qonsole.query" );
+      if (window.location.hostname === "localhost" && !query) {
+        if(confirm('No query found in local storage due to cross-origin blocking, returning you to UKHPI instance. Please check the sessionStorage tab in your browser\'s dev tools for the SPARQL query value.')){
+          window.history.go(-1)
```

Comment:
Doesn't this assume that a user arrived via the homepage and not direct link? Would it not be better to provide the root path?

Reply:
The link from UKHPI is `http://localhost:3000/qonsole?query=_localstore` locally, and `https://hmlr-dev-pres.epimorphics.net/app/qonsole?query=_localstore` remotely so [`window.location.hostname`](https://developer.mozilla.org/en-US/docs/Web/API/URL/hostname) should suffice for local development detection.

Reply:
Updated as discussed, thank you! 🙏